### ISSUE TITLE
Does the current version work with Moodle 3.8?

### ISSUE THREAD

**Issue by weimiao86** [Myddleware/myddleware#265]
Hi,

I am working on the integration of Prestashop 1.7 and Moodle 3.8 with Myddleware 2.5.0e
And I haven't made it work, does the current version work with moodle 3.8?
And is there any kind of tutorial I can follow?
Thanks!

**Comment by weimiao86** [comment]
Here is the prod log:

> [2020-02-23 05:58:54] request.ERROR: Uncaught PHP Exception Symfony\Component\HttpKernel\Exception\NotFoundHttpException: "No route found for "GET /bundles/regle/images/solution/myddlewareapi.png" (from "https://bar.buildarobotk12.com/myddleware/web/rule/panel/")" at /home/buildaro/public_html/bar/myddleware/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/EventListener/RouterListener.php line 137 {"exception":"[object] (Symfony\\Component\\HttpKernel\\Exception\\NotFoundHttpException(code: 0): No route found for \"GET /bundles/regle/images/solution/myddlewareapi.png\" (from \"https://bar.buildarobotk12.com/myddleware/web/rule/panel/\") at /home/buildaro/public_html/bar/myddleware/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/EventListener/RouterListener.php:137, Symfony\\Component\\Routing\\Exception\\ResourceNotFoundException(code: 0):  at /home/buildaro/public_html/bar/myddleware/var/cache/prod/appProdProjectContainerUrlMatcher.php:1024)"} []

**Comment by Myddleware** [comment]
Hi @weimiao86 ,
Hope you had a nice week end. 
The error you get is just an image missing. We will add it in the new hotfix version but it doesn't block Myddleware.
Did you follow this tutorial to connect Myddleware to Moodle ? https://myddleware.github.io/myddleware/#/connectors_credentials?id=moodle

**Comment by weimiao86** [comment]
Yes, I have all the connections work, just don't have a working rule...

![1](https://user-images.githubusercontent.com/7043345/75267814-f3d7f380-57b3-11ea-8c6c-d2c73d3040f9.jpg)
![2](https://user-images.githubusercontent.com/7043345/75267937-1b2ec080-57b4-11ea-90c0-4d23601e0967.jpg)
![3](https://user-images.githubusercontent.com/7043345/75267954-1f5ade00-57b4-11ea-9204-e4bf1e735b4f.jpg)
![4](https://user-images.githubusercontent.com/7043345/75267965-2255ce80-57b4-11ea-85f8-cd064050b2f8.jpg)
![5](https://user-images.githubusercontent.com/7043345/75267976-2550bf00-57b4-11ea-9987-0c5540d51478.jpg)

**Comment by Myddleware** [comment]
Ok thanks for the screenshots.
Coukd you open you data transfer and tell us what is the error Moodle return ?

**Comment by weimiao86** [comment]
![6](https://user-images.githubusercontent.com/7043345/75270514-ef153e80-57b7-11ea-8250-78f92e70a0d1.jpg)
![8](https://user-images.githubusercontent.com/7043345/75270530-f3d9f280-57b7-11ea-993b-f8b0c0a9efb5.jpg)
![9](https://user-images.githubusercontent.com/7043345/75270532-f4728900-57b7-11ea-8af3-04d14355ddbe.jpg)
![10](https://user-images.githubusercontent.com/7043345/75270535-f50b1f80-57b7-11ea-930d-0bf5248e1c12.jpg)
![11](https://user-images.githubusercontent.com/7043345/75270537-f5a3b600-57b7-11ea-8c8d-20151879d75d.jpg)

**Comment by weimiao86** [comment]
![6](https://user-images.githubusercontent.com/7043345/75270514-ef153e80-57b7-11ea-8250-78f92e70a0d1.jpg)
![8](https://user-images.githubusercontent.com/7043345/75270530-f3d9f280-57b7-11ea-993b-f8b0c0a9efb5.jpg)
![9](https://user-images.githubusercontent.com/7043345/75270532-f4728900-57b7-11ea-8af3-04d14355ddbe.jpg)
![10](https://user-images.githubusercontent.com/7043345/75270535-f50b1f80-57b7-11ea-930d-0bf5248e1c12.jpg)
![11](https://user-images.githubusercontent.com/7043345/75270537-f5a3b600-57b7-11ea-8c8d-20151879d75d.jpg)

**Comment by weimiao86** [comment]
And there is no open task in the task list(from page 1 to 50), but I still cannot edit these rules. Is there a way I can reset the Myddleware?
![Screen Shot 2020-02-25 at 10 29 37 AM](https://user-images.githubusercontent.com/7043345/75271717-e02f8b80-57b9-11ea-958c-026c187e8408.jpg)

**Comment by Myddleware** [comment]
Hi,
To edit the rule, you have to cancel (or send them successfully) all data transfer. We can't edit a rule if there is a data transfer open or in error for this rule.
All data transfers in status relate_ko or predecesssor_ko  are not really error. It is just data transfer which are blocked by other data transfer.
The errors that is important to solve are the "error sending" errors. I see 2 cases : 

-  user rule : Moodle indicate that there is a password problem. You can try to change it directly in Myddleware by double clicking on the password field. It could help you to test your new password easily. You can also try to create a user manually in Moodle with this password. Moodle will tell you what is wrong.
- Course rule : There is a category context problem. In the template we have set 1 by default but depending of your Moodle configuration, this id can be different. 

Once you have unlock the data transfer for these 2 rules, the other data transfer should be unlocked. You would just have to click on the button "rerun error" on the rule list view.

**Comment by weimiao86** [comment]
Below is the log from Moodle when I edit my test rules, I tried to create a user from prestashop, however, no new user is created.
![Screen Shot 2020-02-25 at 2 01 45 PM](https://user-images.githubusercontent.com/7043345/75287406-8938af00-57d7-11ea-9d50-a9cfa5b0a25c.jpg)

**Comment by weimiao86** [comment]
Latest log from myddleware task

[2020-02-25 15:15:06] app.ERROR: Failed : Notice: Undefined variable: dataInsert /home/buildaro/public_html/bar/myddleware/src/Myddleware/RegleBundle/Classes/document.php Line : ( 1115 ) [] []

**Comment by Myddleware** [comment]
Great if you were able to edit your rule. Now if you want to read a customer from Prestashop, you have to change the reference date of your rule and set it in the past. Here more info about this reference date : https://myddleware.github.io/myddleware/#/basic_usage?id=rule-parameters

Please let me know if you could read a customer from Prestashop.

**Comment by weimiao86** [comment]
Hi, thanks for your help, I can import users and courses from prestashop to moodle now, still have issues  in orders and student enrollment, I will try to fix it, thank you very much.

**Comment by Myddleware** [comment]
Hi @weimiao86 ,
Great news!
Thanks for your feedback. 
Do not hesitate to open a new issue if you have another problem.
